picatrix/magics/timesketch.py

```diff
@@ -0,0 +1,1235 @@
+"""Timesketch magic module.
+
+This module conatins an implementation for the Picatrix Timesketch integration.
+It enables colab/jupyter to send and receive data from a Timesketch sketch.
+"""
+import datetime
+import logging
+import os
+
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Text
+from typing import Union
+
+import dateutil.parser
+
+from dfdatetime import time_elements
+
+import pandas as pd
+
+from timesketch_api_client import aggregation as api_aggregation
+from timesketch_api_client import config
+from timesketch_api_client import view as api_view
+from timesketch_api_client import sketch as api_sketch
+from timesketch_api_client import story as api_story
+from timesketch_api_client import timeline as api_timeline
+from timesketch_import_client import helper
+from timesketch_import_client import importer
+
+from picatrix.lib import framework
+from picatrix.lib import state
+from picatrix.lib import utils
+
+
+def _fix_return_fields(
+    return_fields: Union[Text, List[Text], None]) -> Optional[Text]:
+  """Returns a fixed string of return fields.
+
+  The Timesketch API expects the return fields to be a comma separated
+  string with no spaces in it and it cannot start with a quotation mark.
+
+  Args:
+    return_fields (str|list): a list of strings or a string with the
+        return fields.
+
+  Returns:
+    A string that can be passed to the Timesketch API.
+  """
+  if isinstance(return_fields, str):
+    return_fields = return_fields.split(',')
+
+  if return_fields is None:
+    return None
+
+  return_field_list = []
+  for field in return_fields:
+    field = field.replace('\'', '')
+    field = field.replace('"', '')
+    field = field.strip()
+
+    return_field_list.append(field)
+
+  return ','.join(return_field_list)
+
+
+def _label_search(
+    label: Text,
+    return_fields: Optional[Text] = '',
+    max_entries: Optional[int] = None) -> pd.DataFrame:
+  """Returns a DataFrame with all events that have a certain label.
+
+  Args:
+    label (str): a string representing the label to search for.
+    return_fields (str): a comma separated string with all the fields that
+        should be returned. If not provided all fields will be returned.
+    max_entries (int): an integer with the maximum number of entries to return.
+
+  Returns:
+      A pandas DataFrame with the search results.
+  """
+  connect()
```

Comment:
I would name this something more telling, like connect_to_timesketch(), but totally up to you.